indigo/indigo/src/main/scala/indigo/shared/events/GlobalEvent.scala

```diff
@@ -947,10 +948,15 @@ object KeyboardEvent {
   *   Reference to the loaded asset
   * @param volume
   *   What volume level to play at
+  * @param switch
+  *   How to handle the previous sounds
   */
-final case class PlaySound(assetName: AssetName, volume: Volume) extends GlobalEvent:
+final case class PlaySound(assetName: AssetName, volume: Volume, switch: SoundSwitch = SoundSwitch.Continue)
+    extends GlobalEvent:
   def withVolume(newVolume: Volume): PlaySound =
     this.copy(volume = newVolume)
+  def withSwitch(newSwitch: SoundSwitch): PlaySound =
+    this.copy(switch = newSwitch)
```

Comment:
Can we remove the default argument, and add something like the below instead? Indigo doesn't employ default arguments anyway. I like to be explicit about valid combinations - which they all are in this case - but it keeps it consistent.

```scala
object PlaySound:

  def apply(assetName: AssetName): PlaySound =
    PlaySound(assetName, Volume.Max, SoundSwitch.Continue)
    
  def apply(assetName: AssetName, volume: Volume): PlaySound =
    PlaySound(assetName, volume, SoundSwitch.Continue)
```

Reply:
Renamed in https://github.com/PurpleKingdomGames/indigo/pull/877/commits/980d8e0093f9abddb244551432259bc33775c7da